torch/csrc/distributed/c10d/symm_mem/nvshmem_extension.cuh

```diff
@@ -1,16 +1,19 @@
 #pragma once
 
+#include <c10/macros/Macros.h>
 #include <ATen/ATen.h>
 
-#include <torch/csrc/distributed/c10d/Store.hpp>
+#define NVSHMEM_CHECK(stmt, msg)                                             \
+  do {                                                                       \
+    int result = (stmt);                                                     \
+    TORCH_CHECK(                                                             \
+        result == 0,                                                         \
+        std::string(__FILE__) + ":" + std::to_string(__LINE__) + " " + msg + \
```

Comment:
Hmm, this string is always constructed where the torch check fails or passes... Reformulate to c10::LazyStr or atleast fmtlib.